scripts/pre_commit_linter.py

```diff
@@ -714,11 +718,142 @@ def _check_spacing(all_files):
     return summary_messages
 
 
+def _check_comments(all_files):
+    """This functions checks for a period at the end of comment."""
+    print 'Starting comment checks'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    message = 'There should be a period at the end of the comment.'
+    failed = False
+    for filename in files_to_check:
+        f = open(filename, 'r')
+        content = f.readlines()
+        length = len(content)
+        line_num = 0
+        while line_num < length:
+            line = content[line_num].lstrip().rstrip()
+            if line.startswith('#'):
+                if line_num + 1 == length:
+                    if line[-1] not in PUNCTUATIONS and line[1:].lstrip()[0].isupper():
+                        failed = True
+                        print '%s --> Line %s: %s' % (
+                            filename, line_num + 1, message)
+                    break                        
+
+                next_line = content[line_num + 1].lstrip().rstrip()
+
+                if not next_line.startswith('#'):
```

Comment:
Instead of repeating code, you might want to just check whether the next line is a comment before checking there is a period.

Reply:
I have modified the code to reduce repetition.